### PR TITLE
Stubout ibm cost 1058

### PR DESCRIPTION
**Pull request by ddonahue007**
Closes: https://issues.redhat.com/browse/COST-1058

Stubbed out IBM views using GCP sources for mocking

When running locally in dev mode you can hit this URL to access IBM
https://ci.foo.redhat.com:1337/beta/cost-management/infrastructure/ibm?group_by[account]=*&order_by[cost]=desc

Screen Shot 2021-03-08 at 1 23 16 PM

Screen Shot 2021-03-08 at 2 24 49 PM

### CODE REVIEW

File: src/api/tags/tag.ts [hunk 1]
```diff
@@ -40,5 +40,6 @@ export const enum TagPathsType {
   aws = 'aws',
   azure = 'azure',
   gcp = 'gcp',
+  ibm = 'ibm',
```
Comment (dlabrecq) [inline src/api/tags/tag.ts:43]: I think this will have to be "gcp" for now. Don't believe the tags API currently supports IBM?

**Review by dlabrecq — CHANGES_REQUESTED** [review]

File: src/components/async/permissionsComponent/permissions.tsx [hunk 2]
```diff
@@ -64,6 +64,7 @@ class PermissionsBase extends React.Component<PermissionsProps> {
     const costModel = userAccess.data.find(d => d.type === UserAccessType.cost_model);
     const explorer = userAccess.data.find(d => d.type === UserAccessType.explorer);
     const gcp = userAccess.data.find(d => d.type === UserAccessType.gcp);
+    const ibm = userAccess.data.find(d => d.type === UserAccessType.gcp);
```
Comment (dlabrecq) [inline src/components/async/permissionsComponent/permissions.tsx:67]: not gcp here

File: src/api/userAccess.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ export const enum UserAccessType {
   cost_model = 'cost_model',
   explorer = 'explorer',
   gcp = 'gcp',
+  ibm = 'ibm',
```
Comment (dlabrecq) [inline src/api/userAccess.ts:24]: this value should be gcp

File: src/pages/costModels/createCostModelWizard/steps.tsx [hunk 4]
```diff
@@ -146,6 +168,12 @@ export const validatorsHash = {
     () => true,
     () => true,
   ],
+  IBM: [
+    ctx => nameErrors(ctx.name) === null && descriptionErrors(ctx.description) === null && ctx.type !== '',
+    ctx => ctx.markup !== '' && !isNaN(Number(ctx.markup)),
+    () => true,
+    () => true,
+  ],
```
Comment (dlabrecq) [inline src/pages/costModels/createCostModelWizard/steps.tsx:176]: I don't think we should update cost models now.

We don't have anyway to hide this from production.

**Review by dlabrecq — CHANGES_REQUESTED** [review]

File: src/pages/views/explorer/explorerHeader.tsx [hunk 5]
```diff
@@ -127,6 +135,9 @@ class ExplorerHeaderBase extends React.Component<ExplorerHeaderProps> {
     if (isGcpAvailable(gcpProviders, gcpProvidersFetchStatus, userAccess)) {
       return PerspectiveType.gcp;
     }
+    if (isIbmAvailable(ibmProviders, ibmProvidersFetchStatus, userAccess)) {
+      return PerspectiveType.gcp;
```
Comment (dlabrecq) [inline src/pages/views/explorer/explorerHeader.tsx:139]: Not gcp here. This should be PerspectiveType.ibm

File: src/api/tags/tag.ts (hunk 1 above)
Comment (ddonahue007) [reply src/api/tags/tag.ts:43]: done

**Review by dlabrecq — CHANGES_REQUESTED** [review]

File: src/store/userAccess/userAccessCommon.ts [hunk 6]
```diff
@@ -28,6 +28,10 @@ export const gcpUserAccessQuery: UserAccessQuery = {
   type: 'GCP',
 };
 
+export const ibmUserAccessQuery: UserAccessQuery = {
+  type: 'IBM',
```
Comment (dlabrecq) [inline src/store/userAccess/userAccessCommon.ts:32]: This will probably need to be 'GCP' in order to see the options in the menu correctly

**Review by dlabrecq — CHANGES_REQUESTED** [review]

File: src/components/async/permissionsComponent/permissions.tsx (hunk 2 above)
Comment (ddonahue007) [reply src/components/async/permissionsComponent/permissions.tsx:67]: Done

File: src/api/userAccess.ts (hunk 3 above)
Comment (ddonahue007) [reply src/api/userAccess.ts:24]: done

File: src/pages/costModels/createCostModelWizard/steps.tsx (hunk 4 above)
Comment (ddonahue007) [reply src/pages/costModels/createCostModelWizard/steps.tsx]: done

File: src/pages/views/explorer/explorerHeader.tsx (hunk 5 above)
Comment (ddonahue007) [reply src/pages/views/explorer/explorerHeader.tsx:139]: done

File: src/api/forecasts/forecast.ts [hunk 7]
```diff
@@ -54,6 +54,7 @@ export const enum ForecastPathsType {
   azure = 'azure',
   azureCloud = 'azure_cloud',
   gcp = 'gcp',
+  ibm = 'ibm',
```
Comment (dlabrecq) [inline src/api/forecasts/forecast.ts:57]: Probably need to change this one to "gcp", too

**Review by dlabrecq — CHANGES_REQUESTED** [review]

**Review by dlabrecq — APPROVED** [review]
LGTM

**Pull request merged**